src/config.rs

```diff
@@ -148,29 +150,37 @@ impl Config {
     }
 
     /// Retrieves all configured auth/delete tokens.
-    pub fn get_tokens(&self, token_type: TokenType) -> Option<Vec<String>> {
-        let mut tokens = match token_type {
+    pub fn get_tokens(&self, token_type: TokenType) -> Option<HashSet<String>> {
+        let tokens = match token_type {
             TokenType::Auth => {
-                let mut tokens = self.server.auth_tokens.clone().unwrap_or_default();
+                let mut tokens: HashSet<_> = self.server.auth_tokens.clone().unwrap_or_default();
+
                 #[allow(deprecated)]
                 if let Some(token) = &self.server.auth_token {
-                    tokens.insert(0, token.to_string());
+                    tokens.insert(token.to_string());
```

Comment:
Is it possible to insert as the first element as before?

Reply:
Btw, what was the point of this? 

I thought it was done this way in order to optimize the access to the vector, but with `HashSet` we have `O(1)` and don't have duplicates 🤔

Reply:
It was just to prioritize checking the tokens, so if auth_token is given via config then it will be checked first etc.

I think this is something trivial though, feel free to ignore.

Reply:
Yeah, I think it's not applicable anymore, we do use `hashset.contains(..)` now, instead of `.iter().any(..)`, so the order doesn't matter actually anymore. Thus, we directly check if token in `Authorization` headers matches any of configured ones with `O(1)` complexity